src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,18 @@
+package duke;
+
+public class Deadline extends Task {
+    String by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    @Override
+    public String getLabel() {
+        String typeIndicator = "[D]";
+        String doneIndicator = "[" + (this.isDone ? "X" : " ") + "]";
```

Comment:
Is the doneIndicator something specific to deadline task or does this also applies to other task types. For the common behaviours, you might want to specify them in the parent class and to use them instead of repeating the same code in every child class.